### PR TITLE
Disable close button when action is in progress

### PR DESCRIPTION
**Pull request by vegaro**
Equivalent of https://github.com/RevenueCat/purchases-ios/pull/3474

### CODE REVIEW

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt [hunk 1]
```diff
@@ -271,6 +271,7 @@ internal class PaywallViewModelImpl(
             validatedPaywallData = displayablePaywall,
             template = template,
             shouldDisplayDismissButton = options.shouldDisplayDismissButton,
+            actionInProgress = actionInProgress.value,
```
Comment (vegaro) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt:274]: I added this here, then realized that we need to update the state in `verifyNoActionInProgressOrStartAction()`, where `actionInProgress` is updated. I am not sure there's a more efficient way of doing it. The alternative would be to update the button wherever the `PaywallListener` is implemented

What do you think @tonidero

Comment (tonidero) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt:274]: Hmm we're already exposing `actionInProgress` from the `PaywallViewModel` as a different `State`. Couldn't we just use that in the `InternalPaywall` directly? I believe in that case this wiring won't be necessary? Do lmk if I'm missing something!

Comment (vegaro) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt]: oh that's right, I missed we were exposing that as a state already

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/LoadingPaywall.kt [hunk 2]
```diff
@@ -110,7 +110,7 @@ private fun LoadingPaywall(
                     ),
             )
         }
-        CloseButton(state.shouldDisplayDismissButton, onDismiss)
+        CloseButton(state.shouldDisplayDismissButton, viewModel.actionInProgress.value, onDismiss)
```
Comment (NachoSoto) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/LoadingPaywall.kt:113]: Maybe add labels to the parameters here and in `InternalPaywall`?

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/composables/CloseButton.kt [hunk 3]
```diff
@@ -10,11 +10,16 @@ import androidx.compose.ui.res.painterResource
...
+fun BoxScope.CloseButton(
+    shouldDisplayDismissButton: Boolean,
+    actionInProgress: Boolean,
+    onClick: () -> Unit,
+) {
     if (shouldDisplayDismissButton) {
         IconButton(
             onClick = onClick,
             modifier = Modifier.align(Alignment.TopStart),
+            enabled = actionInProgress.not(),
```
Comment (NachoSoto) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/composables/CloseButton.kt:22]: So nice

**Review by NachoSoto — APPROVED** [review]
Let's release this?

**Review by tonidero — APPROVED** [review]
Looks great!

**Pull request merged**